src/core_modules/capture-core/components/Dialogs/DiscardDialog.component.js

```diff
@@ -16,9 +18,12 @@ export const ConfirmDialog = ({ open, header, text, confirmText, onConfirm, canc
                 <Button onClick={onCancel} secondary>
                     {cancelText}
                 </Button>
-                <Button onClick={onConfirm} primary>
+                {confirmText ? <Button onClick={onConfirm} primary>
                     {confirmText}
-                </Button>
+                </Button> : null }
+                {destructiveText ? <Button onClick={onDestroy} destructive>
+                    {destructiveText}
+                </Button> : null}
```

Comment:
Is the idea here to incorporate both a `ConfirmDialog` and `DiscardDialog` in the same component? 
In that case, my vote would be to make this a general `Dialog` and pass in a `type`-prop. If the component would strictly be used as a destructive modal, I guess `destructiveText` and `onDestroy` should be required props.